### PR TITLE
UOE-9172: BeforeValidation hook- update the incoming request in OpenW…

### PR DESCRIPTION
**Pull request by AvinashKapre**
…rap module

# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: modules/pubmatic/openwrap/adunitconfig/utils.go [hunk 1]
```diff
@@ -76,12 +77,17 @@ func getFinalSlotAdUnitConfig(slotConfig, defaultConfig *adunitconfig.AdConfig)
 	if (slotConfig.BidFloor == nil || *slotConfig.BidFloor == 0.0) && defaultConfig.BidFloor != nil {
 		slotConfig.BidFloor = defaultConfig.BidFloor
 
-		slotConfig.BidFloorCur = func() *string { s := "USD"; return &s }()
+		slotConfig.BidFloorCur = ptrutil.ToPtr[string](models.USD)
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/adunitconfig/utils.go:80]: use `ptrutil.ToPtr(models.USD)`

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 2]
```diff
@@ -811,15 +809,11 @@ func isSlotEnabled(videoAdUnitCtx, bannerAdUnitCtx models.AdUnitCtx) bool {
...
-	var err error
-
+func getPubID(bidRequest openrtb2.BidRequest) (pubID int, err error) {
 	if bidRequest.Site != nil && bidRequest.Site.Publisher != nil {
 		pubID, err = strconv.Atoi(bidRequest.Site.Publisher.ID)
 	} else if bidRequest.App != nil && bidRequest.App.Publisher != nil {
 		pubID, err = strconv.Atoi(bidRequest.App.Publisher.ID)
 	}
-
 	return pubID, err
```
Comment (pm-nilesh-chate) [inline modules/pubmatic/openwrap/beforevalidationhook.go:818]: you can do a blank `return` here

**Review by pm-nikhil-vaidya — COMMENTED** [review]
LGTM

File: modules/pubmatic/openwrap/abtest_test.go [hunk 3]
```diff
@@ -0,0 +1,591 @@
...
+							models.TestGroupSize + "_test": "0",
+						},
+					},
+				},
+			},
+			want:  nil,
+			want1: false,
+		},
+		{
+			name: "AbTest_is_enabled_and_random_no_return_apply_AbTest",
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/abtest_test.go:110]: Can we add one more condition for `groupsize=90` and `randomvalue>90`?

File: modules/pubmatic/openwrap/abtest_test.go [hunk 4]
```diff
@@ -0,0 +1,591 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			GetRandomNumberIn1To100 = func() int {
+				return tt.args.val
+			}
+			got, got1 := ABTestProcessing(tt.args.rctx)
+			assert.Equal(t, tt.want, got)
+			if got1 != tt.want1 {
+				t.Errorf("ABTestProcessing() got1 = %v, want %v", got1, tt.want1)
+			}
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/abtest_test.go:146]: can we use `assert.Equal` function?

File: modules/pubmatic/openwrap/abtest_test.go [hunk 5]
```diff
@@ -0,0 +1,591 @@
...
+			},
+			want1: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			GetRandomNumberIn1To100 = func() int {
+				return tt.args.val
+			}
+			got, got1 := ABTestProcessing(tt.args.rctx)
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/abtest_test.go:142]: can we rename got and got1 with name like `(config,found)` ?

File: modules/pubmatic/openwrap/abtest_test.go [hunk 6]
```diff
@@ -0,0 +1,591 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			got := UpdateTestConfig(tt.args.rctx)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_copyPartnerConfigMap(t *testing.T) {
+	type args struct {
+		m map[int]map[string]string
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/abtest_test.go:435]: can we rename name for variable to config?

File: modules/pubmatic/openwrap/adapterthrottle_test.go [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+		adapterThrottleMap       map[string]struct{}
+		allPartnersThrottledFlag bool
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "All_prtners_throttled",
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/adapterthrottle_test.go:24]: can we correct partner spelling in other test cases too?

File: modules/pubmatic/openwrap/adapterthrottle_test.go [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			adapterThrottleMap, allPartnersThrottledFlag := GetAdapterThrottleMap(tt.args.partnerConfigMap)
+			assert.Equal(t, tt.want.adapterThrottleMap, adapterThrottleMap)
+			if allPartnersThrottledFlag != tt.want.allPartnersThrottledFlag {
+				t.Errorf("GetAdapterThrottleMap() got1 = %v, want %v", allPartnersThrottledFlag, tt.want.allPartnersThrottledFlag)
+			}
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/adapterthrottle_test.go:101]: please use `assert.Equal` for asserting purpose.

File: modules/pubmatic/openwrap/adapterthrottle_test.go (hunk 7 above)
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/adapterthrottle_test.go:24]: this is case for `all_partner_are_client_side_throttled`.

File: modules/pubmatic/openwrap/adapterthrottle_test.go [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+						models.SERVER_SIDE_FLAG:    "",
+					},
+				},
+			},
+			want: want{
+				adapterThrottleMap:       map[string]struct{}{},
+				allPartnersThrottledFlag: true,
+			},
+		},
+		{
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/adapterthrottle_test.go:46]: Can we add `all_server_side partner_throttled` case too?

File: modules/pubmatic/openwrap/util.go [hunk 10]
```diff
@@ -117,9 +137,16 @@ func GetDevicePlatform(httpReqUAHeader string, bidRequest *openrtb2.BidRequest,
...
-func isMobile(deviceType int, userAgentString string) bool {
-	if deviceType == models.DeviceTypeMobile || deviceType == models.DeviceTypeTablet ||
-		deviceType == models.DeviceTypePhone {
+func isMobile(deviceType adcom1.DeviceType, userAgentString string) bool {
+	if deviceType != 0 {
+		if deviceType == adcom1.DeviceMobile || deviceType == adcom1.DeviceTablet ||
+			deviceType == adcom1.DevicePhone {
+			return true
+		}
+		return false
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/util.go:146]: you can directly return like
`return deviceType == adcom1.DeviceMobile || deviceType == adcom1.DeviceTablet || deviceType == adcom1.DevicePhone`

File: modules/pubmatic/openwrap/util.go [hunk 11]
```diff
@@ -279,3 +302,10 @@ func getPubmaticErrorCode(standardNBR int) int {
 
 	return -1
 }
+
+func isCTV(userAgent string) bool {
+	if ctvRegex.Match([]byte(userAgent)) {
+		return true
+	}
+	return false
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/util.go:310]: you can return directly instead of if condition

File: modules/pubmatic/openwrap/util.go [hunk 12]
```diff
@@ -181,10 +208,6 @@ func getSourceAndOrigin(bidRequest *openrtb2.BidRequest) (string, string) {
 		} else if len(bidRequest.Site.Page) != 0 {
 			source = getDomainFromUrl(bidRequest.Site.Page)
 			origin = source
-			pageURL, err := url.Parse(source)
-			if err == nil && pageURL != nil {
-				origin = pageURL.Host
-			}
```
Comment (PubMatic-OpenWrap) [inline modules/pubmatic/openwrap/util.go:187]: why this is removed, any specific reason?

**Review by PubMatic-OpenWrap — CHANGES_REQUESTED** [review]

File: modules/pubmatic/openwrap/beforevalidationhook_test.go [hunk 13]
```diff
@@ -0,0 +1,1870 @@
...
+				},
+				User: &openrtb2.User{},
+				App: &openrtb2.App{
+					Publisher: &openrtb2.Publisher{
+						ID: "1010",
+					},
+					Content: &openrtb2.Content{
+						Language: "en",
+					},
+				},
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/beforevalidationhook_test.go:1159]: Add missing UTs for fields like `schain`, `wlang` etc

File: modules/pubmatic/openwrap/bidderparams/pubmatic_test.go [hunk 14]
```diff
@@ -0,0 +1,463 @@
...
+				})
+			},
+			want: want{
+				matchedSlot:    "/Test_Adunit1234@Div1@200x300",
+				matchedPattern: "",
+				isRegexSlot:    false,
+				params:         []byte(`{"publisherId":"5890","adSlot":"/Test_Adunit1234@DIV1@200x300","wrapper":{"version":1,"profile":123},"keywords":[{"key":"test_key1","value":["test_value1","test_value2"]},{"key":"test_key2","value":["test_value1","test_value2"]}]}`),
+				wantErr:        false,
+			},
+		},
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/bidderparams/pubmatic_test.go:425]: Include UTs for 

- Pubmatic secondary case 
- When Video object is present in the request and 0x0 is sent in the adSlot field in the pubmatic params
- When native object is present in the request and 1x1 is sent in the adSlot field in the pubmatic params

**Review by pm-isha-bharti — CHANGES_REQUESTED** [review]

File: modules/pubmatic/openwrap/profiledata_test.go [hunk 15]
```diff
@@ -1,12 +1,217 @@
...
+				},
+				-1: {
+					models.PLATFORM_KEY:     models.PLATFORM_APP,
+					models.DisplayVersionID: "1",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "get_profile_data_for_other_than_test_mode_and_cache_returned_error",
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/profiledata_test.go:166]: Rename the test as it is a test request, 
`IsTestRequest: 1` on line no `177`

File: modules/pubmatic/openwrap/util.go (hunk 12 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/util.go:187]: `getDomainFromUrl` method parsing the URL already so no need to parse again. if we do so will get empty `pageURL`. keeping behaviour same to header-bidding.

File: modules/pubmatic/openwrap/abtest_test.go [hunk 16]
```diff
@@ -0,0 +1,591 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := UpdateTestConfig(tt.args.rctx)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_copyPartnerConfigMap(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:433]: Done

File: modules/pubmatic/openwrap/abtest_test.go [hunk 17]
```diff
@@ -0,0 +1,591 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := copyPartnerConfigMap(tt.args.m)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_replaceControlConfig(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:478]: Done

File: modules/pubmatic/openwrap/abtest_test.go [hunk 18]
```diff
@@ -0,0 +1,591 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			replaceControlConfig(tt.args.partnerConfig, tt.args.partnerID, tt.args.key)
+			assert.Equal(t, tt.want, tt.args.partnerConfig)
+		})
+	}
+}
+
+func Test_copyTestConfig(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:538]: Done

File: modules/pubmatic/openwrap/beforevalidationhook_test.go [hunk 19]
```diff
@@ -0,0 +1,1870 @@
...
+	testReq.Cur = []string{}
+
+	testReq.Device = &openrtb2.Device{
+		DeviceType: 1,
+		Language:   "english",
+	}
+	return testReq
+}
+
+func Test_getPageURL(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook_test.go:401]: Done

File: modules/pubmatic/openwrap/beforevalidationhook_test.go [hunk 20]
```diff
@@ -0,0 +1,1870 @@
...
+				},
+			},
+			want: &openrtb2.Imp{
+				ID:     "testImp",
+				Banner: &openrtb2.Banner{},
+				Exp:    10,
+			},
+		},
+		{
+			name: "imp_has_banner_object_but_adunitConfig_banner_is_nil._imp_video_will_not_be_updated",
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook_test.go:1595]: Done

File: modules/pubmatic/openwrap/adunitconfig/regex_test.go [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+import (
+	"testing"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/modules/pubmatic/openwrap/models"
+	"github.com/prebid/prebid-server/modules/pubmatic/openwrap/models/adunitconfig"
+	"github.com/prebid/prebid-server/util/ptrutil"
+)
+
+func Test_getRegexMatch(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adunitconfig/regex_test.go:12]: Done

File: modules/pubmatic/openwrap/abtest_test.go [hunk 22]
```diff
@@ -0,0 +1,591 @@
...
+				t.Errorf("CheckABTestEnabled() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestABTestProcessing(t *testing.T) {
+	type args struct {
+		rctx models.RequestCtx
+		val  int
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:71]: Done

File: modules/pubmatic/openwrap/abtest_test.go (hunk 3 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:110]: Done

File: modules/pubmatic/openwrap/abtest_test.go (hunk 4 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:146]: Done

File: modules/pubmatic/openwrap/abtest_test.go (hunk 5 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:142]: Done

File: modules/pubmatic/openwrap/abtest_test.go (hunk 6 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:435]: Done

File: modules/pubmatic/openwrap/adapterthrottle_test.go (hunk 7 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adapterthrottle_test.go:24]: Done

File: modules/pubmatic/openwrap/adapterthrottle_test.go (hunk 8 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adapterthrottle_test.go:101]: Done

File: modules/pubmatic/openwrap/adapterthrottle_test.go (hunk 7 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adapterthrottle_test.go:24]: Done

File: modules/pubmatic/openwrap/adapterthrottle_test.go (hunk 9 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adapterthrottle_test.go:46]: Done

File: modules/pubmatic/openwrap/util.go (hunk 10 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/util.go:146]: Done

File: modules/pubmatic/openwrap/util.go (hunk 11 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/util.go:310]: Done

File: modules/pubmatic/openwrap/beforevalidationhook_test.go [hunk 23]
```diff
@@ -0,0 +1,1870 @@
...
+			args: args{
+				bidRequest: openrtb2.BidRequest{
+					Site: &openrtb2.Site{
+						Publisher: &openrtb2.Publisher{
+							ID: "5800",
+						},
+					},
+					App: &openrtb2.App{
+						Publisher: &openrtb2.Publisher{
+							ID: "5890",
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook_test.go:1066]: Done

File: modules/pubmatic/openwrap/adapters/bidders.go [hunk 24]
```diff
@@ -559,3 +559,46 @@ func builderApacdex(params BidderParameters) (json.RawMessage, error) {
 	jsonStr.WriteByte('}')
 	return jsonStr.Bytes(), nil
 }
+
+func builderUnruly(params BidderParameters) (json.RawMessage, error) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adapters/bidders.go:563]: Will address this in the removing bidder info and bidder param static file in header-bidding.

File: modules/pubmatic/openwrap/adapterthrottle_test.go [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+	}{
+		{
+			name: "All_prtners_throttled",
+			args: args{
+				partnerConfigMap: map[int]map[string]string{
+					0: {
+						models.THROTTLE:            "0",
+						models.PREBID_PARTNER_NAME: "pubmatic",
+						models.BidderCode:          "pubmatic",
+						models.SERVER_SIDE_FLAG:    "0",
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adapterthrottle_test.go:31]: Added both the cases server-side enabled =0 and 1.

File: modules/pubmatic/openwrap/adunitconfig/utils_test.go [hunk 26]
```diff
@@ -0,0 +1,291 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := getDefaultAllowedConnectionTypes(tt.args.adUnitConfigMap)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_getFinalSlotAdUnitConfig(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adunitconfig/utils_test.go:131]: Done

File: modules/pubmatic/openwrap/adunitconfig/video_test.go [hunk 27]
```diff
@@ -0,0 +1,353 @@
...
+								CompanionType: []adcom1.CompanionType{1, 2, 3},
+							},
+							ConnectionType: []int{10, 20, 30},
+						},
+					},
+				},
+				UsingDefaultConfig:     true,
+				AllowedConnectionTypes: []int{10, 20, 30},
+			},
+		},
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adunitconfig/video_test.go:342]: Added the test case `Video_config_is_prsent_in_both_default_and_slot_preferance_is_given_to_slot_level`

File: modules/pubmatic/openwrap/abtest_test.go [hunk 28]
```diff
@@ -0,0 +1,591 @@
...
+				},
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := CheckABTestEnabled(tt.args.rctx); got != tt.want {
+				t.Errorf("CheckABTestEnabled() = %v, want %v", got, tt.want)
+			}
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/abtest_test.go:63]: Done

File: modules/pubmatic/openwrap/adunitconfig/utils_test.go [hunk 29]
```diff
@@ -0,0 +1,291 @@
...
+		},
+	},
+	Floors: &openrtb_ext.PriceFloorRules{
+		FloorMin:    10,
+		FloorMinCur: "USD",
+		Enabled:     ptrutil.ToPtr(true),
+	},
+}
+
+func Test_getDefaultAllowedConnectionTypes(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adunitconfig/utils_test.go:69]: Done

File: modules/pubmatic/openwrap/beforevalidationhook_test.go (hunk 13 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook_test.go:1159]: Done

File: modules/pubmatic/openwrap/profiledata_test.go (hunk 15 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/profiledata_test.go:166]: Done

File: modules/pubmatic/openwrap/adunitconfig/banner_test.go [hunk 30]
```diff
@@ -0,0 +1,299 @@
...
+											W:  ptrutil.ToPtr[int64](100),
+											H:  ptrutil.ToPtr[int64](200),
+											ID: "123",
+										},
+									},
+								},
+							},
+							"/12344/test_adunit": {
+								Video: &adunitconfig.Video{
+									Enabled: ptrutil.ToPtr(true),
```
[the remark this answers is not in the document]
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/adunitconfig/banner_test.go:180]: Added case `Banner_config_is_prsent_in_both_default_and_slot_preferance_is_given_to_slot_level`

File: modules/pubmatic/openwrap/bidderparams/pubmatic_test.go (hunk 14 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/bidderparams/pubmatic_test.go:425]: Done

Comment (AvinashKapre) [reply modules/pubmatic/openwrap/bidderparams/pubmatic_test.go:441]: Done

**Review by PubMatic-OpenWrap — APPROVED** [review]

File: modules/pubmatic/openwrap/bidderparams/pubmatic_test.go [hunk 31]
```diff
@@ -571,6 +643,207 @@ func TestPreparePubMaticParamsV25(t *testing.T) {
...
+									Values: []string{"test_value1", "test_value2"},
+								},
+							},
+						},
+					},
+					Wrapper: &models.ExtImpWrapper{
+						Div: "Div1",
+					},
+				},
+				imp:       getTestImp("/Test_Adunit1234", false, false),
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/bidderparams/pubmatic_test.go:684]: Is this a native request?

File: modules/pubmatic/openwrap/bidderparams/pubmatic_test.go [hunk 32]
```diff
@@ -571,6 +643,207 @@ func TestPreparePubMaticParamsV25(t *testing.T) {
...
+			want: want{
+				matchedSlot:    "",
+				matchedPattern: "/Test_Adunit1234@200x300",
+				isRegexSlot:    false,
+				params:         []byte(`{"publisherId":"5890","adSlot":"/Test_Adunit1234","wrapper":{"version":1,"profile":123},"keywords":[{"key":"test_key1","value":["test_value1","test_value2"]},{"key":"test_key2","value":["test_value1","test_value2"]}]}`),
+				wantErr:        false,
+			},
+		},
+		{
+			name: "pubmatic_param_for_native_default",
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/bidderparams/pubmatic_test.go:781]: We have two test cases with the same name: `pubmatic_param_for_native_default`. You may want to use one for the video case

**Review by pm-isha-bharti — APPROVED** [review]

**Review by PubMatic-OpenWrap — APPROVED** [review]

**Pull request merged**